src/components/process-dialog.ts

```diff
@@ -24,7 +24,7 @@ export class ESPHomeProcessDialog extends LitElement {
       <mwc-dialog
         open
         heading=${this.heading}
-        scrimClickAction
+        scrimClickAction=""
```

Comment:
I agree this doesn't look pretty, but without this I get this error in VS Code:
![image](https://user-images.githubusercontent.com/1741838/191557962-b89919c3-46b4-4487-8bf7-ffd8f83426dc.png)

same with `lit-analyzer`:

>     Type 'true' is not assignable to 'string'
>     27:  scrimClickAction
>     no-incompatible-type-binding

both properties (scrimClickAction, escapeKeyAction) are of type string:
![image](https://user-images.githubusercontent.com/1741838/191558077-44230ba1-8cd5-4a0d-9a7d-5b4a2bd4dab7.png)

not sure how to solve that better way instead of extending `mwc-dialog` and adding custom properties.

Reply:
More info: https://github.com/material-components/material-components-web/tree/master/packages/mdc-dialog#dialog-actions

Reply:
Attributes are always strings, a string without a value is interpreted as an empty string in HTML. This sounds like a bug in lit-analyzer.

Reply:
Take a look at https://github.com/esphome/dashboard/blob/main/src/components/process-dialog.ts#L25-L27.
In VS Code I see an error for `scrimClickAction`:
![image](https://user-images.githubusercontent.com/1741838/191906328-be62681f-8227-459e-8a81-a72c1701008a.png)

Looking at the definition of mwc-dialog (mwc-dialog-base.d.ts) I see this:

```
/**
 * @license
 * Copyright 2019 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
import 'blocking-elements';
import 'wicg-inert';
import { MDCDialogAdapter } from '@material/dialog/adapter.js';
import MDCDialogFoundation from '@material/dialog/foundation.js';
import { BaseElement } from '@material/mwc-base/base-element.js';
export { MDCDialogCloseEventDetail } from '@material/dialog/types.js';
export declare class DialogBase extends BaseElement {
    protected mdcRoot: HTMLDivElement;
    protected primarySlot: HTMLElement;
    protected secondarySlot: HTMLElement;
    protected contentSlot: HTMLElement;
    protected contentElement: HTMLDivElement;
    protected conatinerElement: HTMLDivElement;
    hideActions: boolean;
    stacked: boolean;
    heading: string;
    scrimClickAction: string;
    escapeKeyAction: string;
    open: boolean;
    defaultAction: string;
    actionAttribute: string;
    initialFocusAttribute: string;
    set suppressDefaultPressSelector(selector: string);
    /**
     * @export
     */
    get suppressDefaultPressSelector(): string;
    protected closingDueToDisconnect?: boolean;
    protected initialSupressDefaultPressSelector: string;
    protected get primaryButton(): HTMLElement | null;
    protected currentAction: string | undefined;
    protected mdcFoundationClass: typeof MDCDialogFoundation;
    protected mdcFoundation: MDCDialogFoundation;
    protected boundHandleClick: ((ev: MouseEvent) => void) | null;
    protected boundHandleKeydown: ((ev: KeyboardEvent) => void) | null;
    protected boundHandleDocumentKeydown: ((ev: KeyboardEvent) => void) | null;
    protected emitNotification(name: string, action?: string): void;
    protected getInitialFocusEl(): HTMLElement | null;
    protected searchNodeTreesForAttribute(nodes: Node[], attribute: string): HTMLElement | null;
    protected createAdapter(): MDCDialogAdapter;
    protected render(): import("lit-html").TemplateResult<1>;
    protected renderHeading(): import("lit-html").TemplateResult<1>;
    protected firstUpdated(): void;
    connectedCallback(): void;
    disconnectedCallback(): void;
    forceLayout(): void;
    focus(): void;
    blur(): void;
    protected setEventListeners(): void;
    protected removeEventListeners(): void;
    close(): void;
    show(): void;
}

```

`open` is a type boolean (no error), but `scrimClickAction` is a type string (we get an error), I think that's the reason for the error.
Saying that I'm not sure if this is an error in the lit-plugin (vs code extensions) or if should this be fixed in the dashboard code base.

Reply:
https://github.com/lit/lit/blob/main/packages/reactive-element/src/reactive-element.ts#L333

Lit's fromAttribute convertor has no case for string, so it treats strings just as they get it from the HTML DOM object. An empty attribute is returned as an empty string. That is implicit. It's a bug in lit analyzer that it does not accept an empty attribute for a string value. It's not that it didn't extract the type correctly.